mobile/src/main/java/org/openhab/habdroid/ui/AbstractItemPickerActivity.kt

```diff
@@ -222,6 +222,11 @@ abstract class AbstractItemPickerActivity : AbstractBaseActivity(), SwipeRefresh
             return
         }
 
+        updateViewVisibility(loading = true, loadError = false, showHint = false)
+        launch {
+            ConnectionFactory.waitForInitialization()
```

Comment:
This does not work. The `launch` block spawns a new thread, which then waits for init, but the code below does not block for that wait to finish. You need to run the code below in main thread context inside this launch block.

Reply:
Something like this will probably do (untested):

```diff
diff --git a/mobile/src/main/java/org/openhab/habdroid/ui/AbstractItemPickerActivity.kt b/mobile/src/main/java/org/openhab/habdroid/ui/AbstractItemPickerActivity.kt
index 20d795be..a2268d9c 100644
--- a/mobile/src/main/java/org/openhab/habdroid/ui/AbstractItemPickerActivity.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/ui/AbstractItemPickerActivity.kt
@@ -222,16 +222,18 @@ abstract class AbstractItemPickerActivity : AbstractBaseActivity(), SwipeRefresh
             return
         }
 
-        val connection = ConnectionFactory.usableConnectionOrNull
-        if (connection == null) {
-            updateViewVisibility(loading = false, loadError = true, showHint = false)
-            return
-        }
-
-        itemPickerAdapter.clear()
         updateViewVisibility(loading = true, loadError = false, showHint = false)
+        itemPickerAdapter.clear()
 
         requestJob = launch {
+            ConnectionFactory.waitForInitialization()
+
+            val connection = ConnectionFactory.usableConnectionOrNull
+            if (connection == null) {
+                updateViewVisibility(loading = false, loadError = true, showHint = false)
+                return@launch
+            }
+
             try {
                 val result = connection.httpClient.get("rest/items").asText()
                 var items = JSONArray(result.response)
```

Reply:
Thanks!